src/Features/Core/Portable/RemoveUnnecessaryImports/AbstractRemoveUnnecessaryImportsDiagnosticAnalyzer.cs

```diff
@@ -47,15 +47,16 @@ private void EnsureClassificationIdDescriptors()
                                              DiagnosticCategory.Style,
                                              DiagnosticSeverity.Hidden,
                                              isEnabledByDefault: true,
-                                             customTags: DiagnosticCustomTags.Unnecessary);
+                                             customTags: new[] { WellKnownDiagnosticTags.Unnecessary, WellKnownDiagnosticTags.Telemetry, WellKnownDiagnosticTags.NotConfigurable });
```

Comment:
Does this mean users can never enforce squiggles in the IDE for unnecessary usings? I presume the project system requires this behavior. They configure [RemoveUnnecessaryImportsDiagnosticId](http://source.roslyn.io/#Microsoft.CodeAnalysis.Features/Diagnostics/Analyzers/IDEDiagnosticIds.cs,11) to be an error over [here](https://github.com/dotnet/project-system/blob/42871fa6e23b766bc50d132f7157b7676f8238a0/build/Default.ruleset#L128): `<Rule Id="IDE0005" Action="Error" />                      <!-- Using directive is unnecessary               using System.Text; `

Reply:
Tagging @davkean too

Reply:
Oh, why is this tied to this change?

Reply:
> Does this mean users can never enforce squiggles in the IDE for unnecessary usings

Yes, but this behavior was already broken anyway. It was never designed to be configurable. See #23695.

Reply:
I read that and still don't understand context. Assume I know nothing about how this works; why is it broken? It works in my limited testing.

Reply:
@mavasani IDE0005 needs two changes before it can be configured:

1. Needs to avoid reporting the errors in generated code (but keep reporting a hidden diagnostic for fading unnecessary code)
2. Needs to be able to run OOP

In the meantime the configuration will be silently ignored, and then when the diagnostic is properly enabled it will become active again.